src/Domain/CoordinateMaps/CubicScale.hpp

```diff
@@ -28,66 +28,78 @@ namespace domain {
 namespace CoordMapsTimeDependent {
 /*!
  * \ingroup CoordMapsTimeDependentGroup
- * \brief CubicScale map defined by \f$x = a(t)*\xi+(b(t)-a(t))\xi^3/X^2\f$,
- *  where \f$X\f$ is the outer boundary.
+ * \brief Maps the radius \f$r\f$ as \f$r(t) = a(t)\rho + \left(b(t) - a(t)
+ * \frac{\rho^3} {R^2}\right)\f$ where \f$\rho\f$ is the radius of the source
+ * coordinates.
  *
- * The map scales the coordinates \f$\xi\f$ near the center by a factor
- * \f$a(t)\f$, while the coordinates near the outer boundary \f$X\f$, are scaled
+ * The map scales the radius \f$\rho\f$ in the source coordinates by a factor
+ * \f$a(t)\f$, while the coordinates near the outer boundary \f$R\f$, are scaled
  * by a factor \f$b(t)\f$. Here \f$a(t)\f$ and \f$b(t)\f$ are FunctionsOfTime.
  *
- * Currently, only a 1-D implementation.
+ * The inverse map is computed by solving the cubic equation:
+ *
+ * \f{align}{
+ * (b-a)\frac{\rho^3}{R^2} + a \rho - r = 0,
+ * \f}
+ *
+ * which is done by defining \f$q=\rho/R\f$, and solving
+ *
+ * \f{align}{
+ * q ((b-a) q^2 + a) - \frac{r}{R} = 0.
+ * \f}
+ *
+ * The source coordinates are obtained using:
+ *
+ * \f{align}{
+ * \xi^i = \frac{qR}{r} x^i(t)
+ * \f}
  */
+template <size_t Dim>
 class CubicScale {
  public:
-  static constexpr size_t dim = 1;
+  static constexpr size_t dim = Dim;
 
-  explicit CubicScale(double outer_boundary) noexcept;
+  explicit CubicScale(double outer_boundary,
+                      std::string function_of_time_name_a,
+                      std::string function_of_time_name_b) noexcept;
   CubicScale() = default;
 
   template <typename T>
-  std::array<tt::remove_cvref_wrap_t<T>, 1> operator()(
-      const std::array<T, 1>& source_coords, double time,
+  std::array<tt::remove_cvref_wrap_t<T>, Dim> operator()(
```

Comment:
In practice, would `T` ever be something other than `double` or `DataVector`? Also, why do you remove const/volatile and reference from T?

Reply:
So far no, but it could be a `float` or a `DataVector<float>`

Because the coordinates are sometimes passed in as a `std::reference_wrapper<DataVector>` and we can't return that